Messaging/Messaging.cs

```diff
@@ -121,10 +121,9 @@ public void Send(Packet packet)
                             UpdateRegressionStatisticsInSourceFile(result);
                         }
 
-                        foreach (var pair in result.Results.Statistics)
-                        {
-                            Log.Trace($"STATISTICS:: {pair.Key} {pair.Value}");
-                        }
+                        var statisticsStr = $"{Environment.NewLine}" +
+                            $"{string.Join(Environment.NewLine,result.Results.Statistics.Select(x => $"STATISTICS:: {x.Key} {x.Value}"))}";
+                        Log.Trace(statisticsStr);
```

Comment:
Interesting to see this made a measurable difference. I guess the main speed up is not invoking the log n times. :+1:

Reply:
Exactly, measurable in the `ms` world. Its never too fast! 🐎